### PR TITLE
KAA-1540: [C SDK]  Removed outdated information from auto generated docs

### PR DESCRIPTION
**Pull request by sparadiuk**

### CODE REVIEW

File: client/client-multi/client-c/src/kaa/Mainpage.dox [hunk 1]
```diff
@@ -147,11 +146,5 @@
             <a href="https://docs.kaaproject.org/display/KAA/Collecting+data+from+endpoints">Collecting data from endpoints</a>
```
Comment (vadimol) [inline client/client-multi/client-c/src/kaa/Mainpage.dox]: Could you also update this link?

**Review by vadimol — COMMENTED** [review]
Overall looks good to me.
See comment.

File: client/client-multi/client-c/src/kaa/Mainpage.dox [hunk 2]
```diff
@@ -35,7 +34,7 @@
         $ make install
     @endcode
 
-    Specific build options are described in the <a href='README'>README</a> and <a href="https://docs.kaaproject.org/display/KAA/Endpoint+platforms+integration">in the official online documentation.</a>
+    Specific platforms' build options are described in the <a href='README'>README</a> and <a href="http://kaaproject.github.io/kaa/docs/latest/Programming-guide/Using-Kaa-endpoint-SDKs/">in the official online documentation.</a>
```
Comment (hunch7) [inline client/client-multi/client-c/src/kaa/Mainpage.dox]: s/Specific platforms' build options/Build options for specific platforms

File: client/client-multi/client-c/src/kaa/Mainpage.dox [hunk 3]
```diff
@@ -123,7 +122,7 @@
     @endcode
 
     @note To use the log collection feature, go through
-    <a href="https://docs.kaaproject.org/display/KAA/Collecting+data+from+endpoints#Collectingdatafromendpoints-Coding">additional initialization steps.</a>
+    <a href="http://kaaproject.github.io/kaa/docs/latest/Programming-guide/Key-platform-features/Data-collection">additional initialization steps.</a>
```
Comment (hunch7) [inline client/client-multi/client-c/src/kaa/Mainpage.dox]:
```
@note If you want to use the log collection feature, see <a href="http://kaaproject.github.io/kaa/docs/latest/Programming-guide/Key-platform-features/Data-collection">Data collection</a> documentation page.  
```

File: client/client-multi/client-c/src/kaa/Mainpage.dox [hunk 4]
```diff
@@ -135,23 +134,6 @@
...
-            <a href="https://docs.kaaproject.org/display/KAA/Collecting+data+from+endpoints">Collecting data from endpoints</a>
-        </li>
-    </ul>
-
- \section demo Demo application
-
-    To better familiarize yourself with Kaa C SDK, you may look at our demo application.
-
-    Find the demo source code in the 'example' directory located in the root of the Kaa project.
+    For more information about functions supported by the Kaa C SDK, see our <a href="http://kaaproject.github.io/kaa/docs/latest/Programming-guide/Key-platform-features/">Key platform features</a>.
```
Comment (hunch7) [inline client/client-multi/client-c/src/kaa/Mainpage.dox]:
```
For more information about Kaa C SDK functionality, see <a href="http://kaaproject.github.io/kaa/docs/latest/Programming-guide/Key-platform-features/">Key platform features</a> documentation page.
```

**Review by hunch7 — APPROVED** [review]

**Review by vadimol — APPROVED** [review]
LGTM

**Pull request merged**